worker-message-prioritization-distribution/src/main/java/com/github/workerframework/workermessageprioritization/redistribution/MessageDistributor.java

```diff
@@ -17,51 +17,159 @@
+            LOGGER.info("Target queue(s) to be created: {}", targetQueuesToBeCreated);
+
+            // Create RabbitMQ channel
+            final Channel channel;
+            try {
+                channel = connection.createChannel();
+            } catch (final IOException e) {
+                throw new RuntimeException("IO Exception encountered trying to create a RabbitMQ channel", e);
+            }
+
+            // Create target queues
+            for (final String targetQueueToBeCreated : targetQueuesToBeCreated) {
+                try {
+                    LOGGER.info(String.format("Creating target queue by calling channel.queueDeclare(%s, %s, %s, %s, %s)",
+                                    targetQueueToBeCreated, targetQueueDurable, targetQueueExclusive, targetQueueAutoDelete,
+                            targetQueueArgs));
+
+                    channel.queueDeclare(
+                            targetQueueToBeCreated,
+                            targetQueueDurable,
+                            targetQueueExclusive,
+                            targetQueueAutoDelete,
+                            targetQueueArgs);
+                } catch (final IOException e) {
+                    throw new RuntimeException(String.format(
+                            "IOException thrown trying to create target queue when calling channel.queueDeclare(%s, %s, %s, %s, %s)",
+                            targetQueueToBeCreated, targetQueueDurable, targetQueueExclusive, targetQueueAutoDelete, targetQueueArgs), e);
+                }
+            }
+
+            // Close RabbitMQ channel
+            try {
+                channel.close();
+            } catch (final IOException | TimeoutException e) {
+                LOGGER.warn("Exception thrown trying to close RabbitMQ channel", e);
+            }
+
+            // Return the updated set of target queues
+            return queuesApi.getQueues()
+                    .stream()
+                    .filter(q -> !q.getName().contains(LOAD_BALANCED_INDICATOR))
+                    .collect(Collectors.toSet());
+        } else {
+            // No target queues needed to be created
+            return targetQueuesAlreadyCreated;
+        }
+    }
+
+    public Set<String> getTargetQueuesToBeCreated(final List<Queue> queues, final Set<Queue> targetQueuesAlreadyCreated)
+    {
+        // All staging and target queues returned by the RabbitMQ API
+        final Set<String> targetQueuesAndStagingQueues = queues
+                .stream()
+                .map(Queue::getName)
+                .collect(Collectors.toSet());
+
+        // Parse target queues from staging queues
+        final Set<String> targetQueuesParsedFromStagingQueues = targetQueuesAndStagingQueues
+                .stream()
+                .filter(name -> name.contains(LOAD_BALANCED_INDICATOR))
+                .map(name -> name.split(LOAD_BALANCED_INDICATOR)[0])
+                .collect(Collectors.toSet());
```

Comment:
You could avoid the collecting to a set and then re-streaming by going straight into the filter.

```suggestion
        // Parse target queues from staging queues
        final Set<String> targetQueuesParsedFromStagingQueues = queues
                .stream()
                .map(Queue::getName)
                .filter(name -> name.contains(LOAD_BALANCED_INDICATOR))
                .map(name -> name.split(LOAD_BALANCED_INDICATOR)[0])
                .collect(Collectors.toSet());
```